### PR TITLE
Batching2

### PR DESCRIPTION
**Pull request by jamesturner246**
Use new GeNN batching.

### CODE REVIEW

**Review by neworderofjamie — COMMENTED** [review]
This looks really good apart from a very few tiny things - so glad this didn't prove _too_ annoying 😄

File: ml_genn/layers/input_layer.py [hunk 1]
```diff
@@ -5,11 +5,26 @@
...
-        if not isinstance(self.neurons, InputNeurons):
+        if not isinstance(neurons, InputNeurons):
             raise ValueError('"InputLayer" class instances require "InputNeuron" class neurons')
 
+        super(InputLayer, self).__init__(name, neurons)
+        self.shape = shape
+
     def set_input_batch(self, data_batch):
-        self.neurons.set_input_batch(data_batch)
+        input_view = self.neurons.nrn.vars['input'].view
```
[the remark this answers is not in the document]
Comment (jamesturner246) [reply ml_genn/layers/input_layer.py:15]: ? I have batch dim first, like in TensorFlow, and it seems to work fine.

[the remark this answers is not in the document]
Comment (neworderofjamie) [reply ml_genn/layers/input_layer.py:15]: Oh yeah, sorry I was getting it backwards - this is 2D and obviously the per-batch data is contiguous 🤦

[the remark this answers is not in the document]
Comment (jamesturner246) [reply ml_genn/layers/input_layer.py:15]: phew!

File: ml_genn/layers/base_synapses.py [hunk 2]
```diff
@@ -22,30 +20,20 @@ def set_weights(self, weights):
...
+        name = '{}_to_{}_syn'.format(self.source.name, self.target.name)
+        pre = self.source.neurons.nrn
+        post = self.target.neurons.nrn
+
+        self.syn = mlg_model.g_model.add_synapse_population(
+            name, conn, delay, pre, post,
+            wu_model, wu_params, wu_vars, wu_pre_vars, wu_post_vars,
+            ps_model, ps_params, ps_vars, conn_init)
+
+        for wu_var, wu_var_egp in zip(wu_vars_egp.keys(), wu_vars_egp.values()):
```
[the remark this answers is not in the document]
Comment (jamesturner246) [reply ml_genn/layers/base_synapses.py:37]: I couldn't get this to work.. :(

```
ValueError: not enough values to unpack (expected 2, got 1)
```

**Review by neworderofjamie — APPROVED** [review]

**Pull request merged**